blueocean-core-js/src/js/utils/UrlUtils.js

```diff
@@ -34,7 +34,10 @@ export const buildPipelineUrl = (organization, fullName, tabName) => {
 
     return tabName ? `${baseUrl}/${tabName}` : baseUrl;
 };
-
+export const rootPathCreate = () => {
```

Comment:
We had a lot of problems with different pages building URL's and having logic / encoding bugs in them, so we started to centralize things. We also needed to share logic between dashboard and personalization. 

Once we hit extensibility 2.0 I could see this stuff moving back into Dashboard so that Personalization can just depend on it and use it, although arguably URL's pointing to classic aren't really tied to any Blue Ocean UI module in particular.

It might be sensible to split all of the functions that build URL's to Classic into their own file, so it's sort of an API responsible for making sure we're point to the right area in Classic. That would at least help trim things down a bit.

I have some ideas on how to improve our URL building but probably won't be able to look at it until later in the summer.